packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -378,38 +368,43 @@ export class CapabilityService {
     if (!customer || !customer.subscriptions!.data) {
       return [];
     }
-    const subscribedProducts = customer
+    const subscribedPrices = customer
       .subscriptions!.data.filter((sub) =>
         ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status)
       )
       .flatMap((sub) => sub.items.data)
-      .map(({ price: { product: productId } }) => productId as string);
-    return subscribedProducts;
+      .map(({ price: { id: priceId } }) => priceId as string);
+    return subscribedPrices;
   }
 
   /**
-   * Fetch the list of capabilities for the given product ids.
+   * Fetch the list of capabilities for the given price ids.
    */
-  private async productIdsToClientCapabilities(
-    subscribedProducts: string[]
+  private async priceIdsToClientCapabilities(
+    subscribedPrices: string[]
   ): Promise<ClientIdCapabilityMap> {
     const allCapabilities: Record<string, Set<string>> = {};
-
     // Run through all plans and collect capabilities for subscribed products
-    const plans = await this.stripeHelper.allAbbrevPlans();
-    for (const plan of plans) {
-      if (!subscribedProducts.includes(plan.product_id)) {
+    const prices = await this.stripeHelper.allAbbrevPlans();
+    for (const price of prices) {
+      if (!subscribedPrices.includes(price.plan_id)) {
```

Comment:
Just talking to myself here... but this reminds me of the "archived" (soft delete) plans we discussed recently.  We definitely cannot remove archived plans from our cached list.